### DISCUSSION TITLE
authentication trouble to remote host

### DISCUSSION THREAD

**Discussion by dreyna01** [StackStorm/st2#5694] [Q&A]
Hello all,

I'm new to Stackstorm, I'm getting this error when trying to run linux.check_loadavg from the Stackstorm server to a remote host or to itself. Any advice would be greatly appreciated. Thanks in advance

st2 --version
st2 3.7.0, on Python 3.8.12

failed connecting to host q must be exactly 160, 224, or 256 bits long

and
failed: true
    return_code: 127
    stderr: 'bash: /tmp/62e80e250e6a518eabd95ff4/check_loadavg.py: No such file or directory'
    stdout: ''
    succeeded: false

**Comment by arm4b** [comment]
Are the other linux actions via st2 running OK on that host?

**Comment by dreyna01** [comment]
hi, thank you for the reply. None of the linux actions succeed.

**Comment by dreyna01** [comment]
actually, linux.vmstat succeeded successfully. 

id: 62eadf6e41d5b8bf495e3be2
action.ref: linux.vmstat
context.user: st2admin
parameters:
  hosts: xxx.xx.x.xxx
status: succeeded
start_timestamp: Wed, 03 Aug 2022 20:49:50 UTC
end_timestamp: Wed, 03 Aug 2022 20:49:51 UTC
result:
  xxx.xx.x.xxx:
    failed: false
    return_code: 0
    stderr: ''
    stdout: "procs -----------memory---------- ---swap-- -----io---- -system-- ------cpu-----
 r  b   swpd   free   buff  cache   si   so    bi    bo   in   cs us sy id wa st
 0  0      0 2217468   4204 1428724    0    0     0     2   29    4  0  0 100  0  0"
    succeeded: true

**Comment by dreyna01** [comment]
I got this resolved. First it was a cryptography issue with paramiko on the stackstorm server. Not sure if it was the pip3 install paramiko or the pip3 update to paramkio that fixed that one. Second, it was an sftp issue. Had to change a line in /etc/ssh/sshd_config from Subsystem      sftp    /usr/lib/openssh/sftp-server to Subsystem       sftp    /usr/libexec/openssh/sftp-server. Third, i did a pip3 install st2client on the remote host I was trying to run check_loadavg against. Lastly, created a python link in /opt/stackstorm/st2/bin/ on the remote host. I forgot this but I think before all of this, I had to do a pip3 install orjson on the stackstorm server

**Comment by dreyna01** [comment]
I suppose I should mention it really helped to go through the st2actionrunner logs in /var/log/st2

**Comment by dreyna01** [comment]
[root]# st2 run linux.check_loadavg hosts='xxx.xx.x.xxx'
.
id: 62ed572a61b0ae9630455b31
action.ref: linux.check_loadavg
context.user: st2admin
parameters:
  hosts: xxx.xx.x.xxx
status: succeeded
start_timestamp: Fri, 05 Aug 2022 17:45:14 UTC
end_timestamp: Fri, 05 Aug 2022 17:45:15 UTC
result:
  xxx.xx.x.xxx:
    failed: false
    return_code: 0
    stderr: ''
    stdout:
      '1': '0.0'
      '15': '0.0'
      '5': '0.0'
    succeeded: true
